loader.py

```diff
@@ -1,3 +1,182 @@
+import logging
+from os import walk
+from os.path import isdir, join, splitext
+from util import *
+
+logger = logging.getLogger(__name__)
+
+"""
+interface Organizer:
+
+    __init__(input_file=None)
+        Organizer can use labels from input_file
+        to create mappings into output_dir for
+        records.
+
+    get_dest_paths(record_files, output_dir)
+        Return dict of destination file locations
+        for record files in the output_dir.
+"""
+
+"""
+Mandatory keys in mapping dict.
+"""
+MAPPING_PROPS = {'old_path', 'new_path'}
+
+
+class DateBasedOrganizer:
+    """
+    Organize records (create mappings) based on
+    record creation time.
+    """
+
+    class Record:
+        """Util struct for working with record data."""
+        def __init__(self, filename, creation_time, duration):
+            self.filename = filename
+            self.creation_time = creation_time
+            self.duration = duration
+            self.finish_time = creation_time + duration
+
+        def overlap_with(self, record):
+            return ((self.creation_time < record.finish_time) and
+                    (record.creation_time < self.finish_time))
+
+    def __init__(self, input_file=None):
+        # Input file will be needed for annotations
+        # self.input_file = input_file
+        pass
+
+    def _create_sorted_records(self, record_filenames):
+        """
+        Create list of DateBasedOrganizer.Record objects
+        sorted by creation date.
+        """
+        records = []
+        for filename in record_filenames:
+            try:
+                creation_time = get_encoded_date_in_sec(filename)
+            except NoEncodedDateException:
+                logger.warning('Using \"creation / last modified '
+                               'time\" for file \'%s\'!\n'
+                               'Reason: The file does not contain \"Encoded'
+                               ' Date\" metadata!\n')
+                creation_time = get_creation_time_in_sec(filename)
+            except FileNotFoundError as e:
+                logger.error(str(e))
+                creation_time = None
+            finally:
+                if creation_time is not None:
+                    record = self.Record(filename, creation_time,
+                                         get_duration_in_sec(filename))
+                    records.append(record)
+
+        records = sorted(records, key=lambda rec: rec.creation_time)
+
+        return records
+
+    def _group_records(self, records):
+        """
+        Group records into groups based on recording
+        time overlap.
+        :param records: List of DateBasedOrganizer.Record
+                        objects.
+        :return: List of lists of records.
+        """
+        record_groups = []
+        if len(records) == 0:
+            return record_groups
+        active_record = records.pop(0)
```

Comment:
pls daj koment k tomuto, ze to je kvoli prvemu recordu a tak isto na konci, ze appendujes poslednu active_group. lebo je to trosku zmatocne na prvy pohlad.